### PR TITLE
Door type extension and room fix

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

**Comment by Thundermaker300** [comment]
yamato i swear if you target master one more time

**Comment by Thundermaker300** [comment]
Replace all cases of "room" with "door" in DoorExtensions.

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    /// A set of extensions for <see cref="DoorType"/>.
+    /// </summary>
+    public static class DoorTypeExtensions
+    {
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a gate.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a gate or not.</returns>
+        public static bool IsGate(this DoorType room) => room is DoorType.GateA or DoorType.GateB or DoorType.Scp914Gate or
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:24]:
```suggestion
        public static bool IsGate(this DoorType door) => door is DoorType.GateA or DoorType.GateB or DoorType.Scp914Gate or
```

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            DoorType.Scp049Gate or DoorType.GR18Gate or DoorType.SurfaceGate or DoorType.Scp173Gate;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a checkpoint.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a checkpoint or not.</returns>
+        public static bool IsCheckpoint(this DoorType room) => room is DoorType.CheckpointLczA
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:32]:
```suggestion
        public static bool IsCheckpoint(this DoorType door) => door is DoorType.CheckpointLczA
```

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            or DoorType.CheckpointLczB or DoorType.CheckpointEzHczA or DoorType.CheckpointEzHczB;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a elevator.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a elevator or not.</returns>
+        public static bool IsElevator(this DoorType room) => room is DoorType.ElevatorGateA or DoorType.ElevatorGateB
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:40]:
```suggestion
        public static bool IsElevator(this DoorType door) => door is DoorType.ElevatorGateA or DoorType.ElevatorGateB
```

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a checkpoint or not.</returns>
+        public static bool IsCheckpoint(this DoorType room) => room is DoorType.CheckpointLczA
+            or DoorType.CheckpointLczB or DoorType.CheckpointEzHczA or DoorType.CheckpointEzHczB;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a elevator.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:38]:
```suggestion
        /// <param name="door">The door to be checked.</param>
```

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        /// <returns>Returns whether the <see cref="DoorType"/> is a gate or not.</returns>
+        public static bool IsGate(this DoorType room) => room is DoorType.GateA or DoorType.GateB or DoorType.Scp914Gate or
+            DoorType.Scp049Gate or DoorType.GR18Gate or DoorType.SurfaceGate or DoorType.Scp173Gate;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a checkpoint.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:30]:
```suggestion
        /// <param name="door">The door to be checked.</param>
```

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    /// <summary>
+    /// A set of extensions for <see cref="DoorType"/>.
+    /// </summary>
+    public static class DoorTypeExtensions
+    {
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a gate.
+        /// </summary>
+        /// <param name="room">The door to be checked.</param>
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:22]:
```suggestion
        /// <param name="door">The door to be checked.</param>
```

**Review by Thundermaker300 — CHANGES_REQUESTED** [review]

**Review by Thundermaker300 — APPROVED** [review]

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a checkpoint.
+        /// </summary>
+        /// <param name="door">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a checkpoint or not.</returns>
+        public static bool IsCheckpoint(this DoorType door) => door is DoorType.CheckpointLczA
+            or DoorType.CheckpointLczB or DoorType.CheckpointEzHczA or DoorType.CheckpointEzHczB;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a elevator.
```
Comment (Thundermaker300) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:36]:
```suggestion
        /// Checks if a <see cref="DoorType">door type</see> is an elevator.
```
English :trollface:

**Review by Thundermaker300 — APPROVED** [review]

File: Exiled.API/Extensions/DoorTypeExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        public static bool IsGate(this DoorType door) => door is DoorType.GateA or DoorType.GateB or DoorType.Scp914Gate or
+            DoorType.Scp049Gate or DoorType.GR18Gate or DoorType.SurfaceGate or DoorType.Scp173Gate;
+
+        /// <summary>
+        /// Checks if a <see cref="DoorType">door type</see> is a checkpoint.
+        /// </summary>
+        /// <param name="door">The door to be checked.</param>
+        /// <returns>Returns whether the <see cref="DoorType"/> is a checkpoint or not.</returns>
+        public static bool IsCheckpoint(this DoorType door) => door is DoorType.CheckpointLczA
+            or DoorType.CheckpointLczB or DoorType.CheckpointEzHczA or DoorType.CheckpointEzHczB;
```
Comment (iopietro) [inline Exiled.API/Extensions/DoorTypeExtensions.cs:33]:
```suggestion
        public static bool IsCheckpoint(this DoorType door) => door is DoorType.CheckpointLczA or DoorType.CheckpointLczB or DoorType.CheckpointEzHczA or DoorType.CheckpointEzHczB;
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by iopietro — APPROVED** [review]

**Pull request merged**